lib/galaxy/authnz/psa_authnz.py

```diff
@@ -404,12 +404,14 @@ def disconnect(name=None, user=None, user_storage=None, strategy=None,
     Additionally, returning any value except for a(n) (empty) dictionary, will break the
     disconnect pipeline, and that value will be returned as a result of calling the `do_disconnect` function.
     """
-    user_authnz = strategy.trans.sa_session.query(user_storage).filter(user_storage.table.c.user_id == user.id,
+
+    sa_session = user_storage.sa_session
+    user_authnz = sa_session.query(user_storage).filter(user_storage.table.c.user_id == user.id,
                                                                        user_storage.table.c.provider == name).first()
```

Comment:
There's a linting error in https://app.circleci.com/jobs/github/galaxyproject/galaxy/56933,
you'll want to put the 2nd argument to start at the same column as the first argument:
```suggestion
                                                        user_storage.table.c.provider == name).first()
```
(this is kind-of approximate in the github UI)